tools/js2c.cc

```diff
@@ -381,18 +360,45 @@ std::string GetVariableName(const std::string& id) {
   return result;
 }
 
-std::vector<std::string> GetCodeTable() {
-  size_t size = 1 << 16;
-  std::vector<std::string> code_table(size);
-  for (size_t i = 0; i < size; ++i) {
-    code_table[i] = std::to_string(i) + ',';
+
+
+constexpr std::pair<std::array<char, 382106>,std::array<uint32_t, 65537>> precompute_string() {
```

Comment:
nit: can you add a comment explaining where these numbers come from and why.

Also, given that they are duplicated in a couple of places, defining a constant would be helpful (and safer)